jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -148,6 +149,10 @@ private Session connect(ConfiguredEndpoint configuredEndpoint, URI destURI) thro
                 return sessionFuture.get(timeout+1000, TimeUnit.MILLISECONDS);
             return sessionFuture.get();
         }
+        catch (ExecutionException e)
+        {
+            throw new IOException("Connection future not completed " + destURI, e.getCause());
+        }
         catch (TimeoutException e)
         {
             throw new IOException("Connection future not completed " + destURI, e);
```

Comment:
Please add the timeout value to this message, although it should not happen.

There also is a catch-all `catch (Exception x)` that you want to convert to a `catch (Throwable x)`.

Reply:
Wouldn't the nested cause `TimeoutException` include the timeout value?
I'm not sure we can even dig into the timeout details at this point in time.

Reply:
The `TimeoutException` would be thrown by line 149, which should never happen, so adding the timeout to the exception message is only for "just in case".